File: src/test/java/leets/leenk/domain/feed/domain/service/CommentSaveServiceTest.java
package leets.leenk.domain.feed.domain.service;

import leets.leenk.domain.feed.domain.entity.Comment;
import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.feed.domain.repository.CommentRepository;
import leets.leenk.domain.feed.test.CommentTestFixture;
import leets.leenk.domain.feed.test.FeedTestFixture;
import leets.leenk.domain.feed.test.UserTestFixture;
import leets.leenk.domain.user.domain.entity.User;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.BDDMockito.then;

@ExtendWith(MockitoExtension.class)
public class CommentSaveServiceTest {
    @Mock
    private CommentRepository commentRepository;

    @InjectMocks
    private CommentSaveService commentSaveService;

    @Test
    @DisplayName("saveComment 테스트")
    void saveComment() {
        // given
        User user = UserTestFixture.createUser(1L, "me");
        Feed feed = FeedTestFixture.createFeed(1L, user);
        Comment comment = CommentTestFixture.createComment(null, user, feed, "hi");

        // when
        commentSaveService.saveComment(comment);

        // then
        then(commentRepository).should().save(comment);
    }
}

File: src/test/kotlin/leets/leenk/domain/feed/domain/service/CommentSaveServiceTest.kt
package leets.leenk.domain.feed.domain.service

import io.kotest.core.spec.style.BehaviorSpec
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import leets.leenk.domain.feed.domain.repository.CommentRepository
import leets.leenk.domain.feed.test.CommentTestFixture
import leets.leenk.domain.feed.test.FeedTestFixture
import leets.leenk.domain.feed.test.UserTestFixture

class CommentSaveServiceTest :
    BehaviorSpec({
        val commentRepository = mockk<CommentRepository>(relaxed = true)
        val commentSaveService = CommentSaveService(commentRepository)

        Given("사용자, 피드, 댓글이 주어졌을 때") {
            val user = UserTestFixture.createUser(1L, "me")
            val feed = FeedTestFixture.createFeed(1L, user)
            val comment = CommentTestFixture.createComment(null, user, feed, "hi")

            every { commentRepository.save(any()) } returns comment

            When("댓글을 저장하면") {
                commentSaveService.saveComment(comment)

                Then("commentRepository의 save 메서드가 호출되어야 한다") {
                    verify { commentRepository.save(comment) }
                }
            }
        }
    })
